### PR TITLE
MBMS-43487 | Self-Service: Pre-Need Integration: Applicant and Sponsor Information Other Comment Box

### PR DESCRIPTION
**Pull request by jordanBoardmanCodes**
## Summary

As a Veteran and/or family member, I need the ability to enter my correct race if it is not listed so that I can provide the most accurate race and obtain a quick determination on possible eligibility.

## Related issue(s)

Jira Ticket: https://jira.devops.va.gov/browse/MBMS-43487

## Screenshots

https://github.com/department-of-veterans-affairs/vets-website/assets/155088633/51d85e08-ba0e-403f-abd4-a96dc951342f

https://github.com/department-of-veterans-affairs/vets-website/assets/155088633/e2eb5a94-4f06-40f5-85ce-c8cb7a8672b0

https://github.com/department-of-veterans-affairs/vets-website/assets/155088633/186c5591-6992-4dab-82c0-b2acfddb3a03

https://github.com/department-of-veterans-affairs/vets-website/assets/155088633/b46f44a4-5e90-448e-bdb9-763acfaa7c13

https://github.com/department-of-veterans-affairs/vets-website/assets/155088633/2e22cb38-bd26-48dd-9bfd-dc277318a36d

## What areas of the site does it impact?

Pre-Need Integration

## Acceptance criteria

Screenshot 2024-05-09 at 11 12 00 AM

Screenshot 2024-05-09 at 11 12 50 AM

### Quality Assurance & Testing

- [x] POR/UI/UX Review
- [x] Unit Tests Passing
- [ ] QA
- [x] Dev Approvals

### CODE REVIEW

**Review by ksantiagoBAH — CHANGES_REQUESTED** [review]

File: src/applications/pre-need-integration/config/form.jsx [hunk 1]
```diff
@@ -87,10 +89,10 @@ import {
   applicantDemographicsMaritalStatusTitle,
   applicantDemographicsPreparerGenderTitle,
   applicantDemographicsPreparerMaritalStatusTitle,
-  applicantDemographicsEthnicityTitle,
-  applicantDemographicsRaceTitle,
-  applicantDemographicsPreparerEthnicityTitle,
-  applicantDemographicsPreparerRaceTitle,
+  // applicantDemographicsEthnicityTitle,
```
[the remark this answers is not in the document]
Comment (jordanBoardmanCodes) [reply src/applications/pre-need-integration/config/form.jsx:92]: Done

File: src/applications/pre-need-integration/config/form.jsx [hunk 2]
```diff
@@ -87,10 +89,10 @@ import {
...
+  // applicantDemographicsRaceTitle,
+  // applicantDemographicsPreparerEthnicityTitle,
+  // applicantDemographicsPreparerRaceTitle,
```
[the remark this answers is not in the document]
Comment (jordanBoardmanCodes) [reply src/applications/pre-need-integration/config/form.jsx:95]: Done

File: src/applications/pre-need-integration/utils/helpers.js [hunk 3]
```diff
@@ -873,6 +873,22 @@ export const veteranUI = {
       showFieldLabel: true,
     },
   },
+  raceComment: {
+    'ui:title': 'Enter the race that best describes you',
+    'ui:widget': 'textarea',
+    'ui:required': form => {
+      return form?.application?.veteran?.race?.isOther; // Only triggers the required if the user clicks Other and submits from there
```
[the remark this answers is not in the document]
Comment (jordanBoardmanCodes) [reply src/applications/pre-need-integration/utils/helpers.js:880]: Done

File: src/applications/pre-need-integration/utils/helpers.js [hunk 4]
```diff
@@ -873,6 +873,22 @@ export const veteranUI = {
...
+    },
+    'ui:options': {
+      expandUnder: 'race',
+      maxLength: 100,
+      pattern: /^(?!\s+$)[\w\s.,'"!?()-]+$/, // Checks to make sure the user doesn't input only spaces or any unusual characters
```
[the remark this answers is not in the document]
Comment (jordanBoardmanCodes) [reply src/applications/pre-need-integration/utils/helpers.js:885]: Done

File: src/applications/pre-need-integration/utils/helpers.js [hunk 5]
```diff
@@ -873,6 +873,22 @@ export const veteranUI = {
...
+      hideIf: form => {
+        return !form?.application?.veteran?.race?.isOther; // Safely return the value of isOther
```
[the remark this answers is not in the document]
Comment (jordanBoardmanCodes) [reply src/applications/pre-need-integration/utils/helpers.js:887]: Done

File: src/applications/pre-need-integration/config/form.jsx (hunk 2 above)
Comment (ksantiagoBAH) [inline src/applications/pre-need-integration/config/form.jsx:95]: They are still there it appears

Comment (jordanBoardmanCodes) [reply src/applications/pre-need-integration/config/form.jsx:95]: Whoops yep I had a push issue - one sec

**Review by ksantiagoBAH — DISMISSED** [review]

**Review by ksantiagoBAH — DISMISSED** [review]

**Review by ConnorJeff — DISMISSED** [review]

**Review by delli-gatti-michael — APPROVED** [review]

**Review by delli-gatti-michael — APPROVED** [review]

**Review by ksantiagoBAH — APPROVED** [review]

**Pull request merged**